File: Examples/VisualBasic/Shapes/EmbeddedVideoFrame.vb
Imports System
Imports Microsoft.VisualBasic
Imports System.IO

Imports Aspose.Slides
Imports Aspose.Slides.Export

Namespace Aspose.Slides.Examples.VisualBasic.Shapes
    Public Class EmbeddedVideoFrame
        Public Shared Sub Run()
			'ExStart:EmbeddedVideoFrame
            ' The path to the documents directory.
            Dim dataDir As String = RunExamples.GetDataDir_Shapes()
            Dim videoDir As String = RunExamples.GetDataDir_Video()

            ' Create directory if it is not already present.
            Dim IsExists As Boolean = System.IO.Directory.Exists(dataDir)
            If Not IsExists Then
                System.IO.Directory.CreateDirectory(dataDir)
            End If
            ' Instantiate Presentation class that represents the PPTX
            Using pres As New Presentation()

                ' Get the first slide
                Dim sld As ISlide = pres.Slides(0)

                ' Embedd vide inside presentation
                Dim vid As IVideo = pres.Videos.AddVideo(New FileStream(videoDir & Convert.ToString("Wildlife.mp4"), FileMode.Open))

                ' Add Video Frame
                Dim vf As IVideoFrame = sld.Shapes.AddVideoFrame(50, 150, 300, 350, vid)

                ' Set video to Video Frame
                vf.EmbeddedVideo = vid

                ' Set Play Mode and Volume of the Video
                vf.PlayMode = VideoPlayModePreset.Auto
                vf.Volume = AudioVolumeMode.Loud

                'Write the PPTX file to disk
                pres.Save(dataDir & Convert.ToString("VideoFrame_out.pptx"), SaveFormat.Pptx)
            End Using
			'ExEnd:EmbeddedVideoFrame
        End Sub
    End Class
End Namespace

File: Examples/CSharp/Shapes/EmbeddedVideoFrame.cs
using System.IO;

using Aspose.Slides;
using Aspose.Slides.Export;

namespace Aspose.Slides.Examples.CSharp.Shapes 
{
    public class EmbeddedVideoFrame
    {
        public static void Run()
        {
            //ExStart:EmbeddedVideoFrame
            // The path to the documents directory.
            string dataDir = RunExamples.GetDataDir_Shapes();
            string videoDir = RunExamples.GetDataDir_Video();
            string resultPath = Path.Combine(RunExamples.OutPath, "VideoFrame_out.pptx");

            // Create directory if it is not already present.
            bool IsExists = System.IO.Directory.Exists(dataDir);
            if (!IsExists)
                System.IO.Directory.CreateDirectory(dataDir);
            // Instantiate Presentation class that represents the PPTX
            using (Presentation pres = new Presentation())
            {

                // Get the first slide
                ISlide sld = pres.Slides[0];

                // Embedd vide inside presentation
                IVideo vid = pres.Videos.AddVideo(new FileStream(videoDir + "Wildlife.mp4", FileMode.Open), LoadingStreamBehavior.ReadStreamAndRelease);

                // Add Video Frame
                IVideoFrame vf = sld.Shapes.AddVideoFrame(50, 150, 300, 350, vid);

                // Set video to Video Frame
                vf.EmbeddedVideo = vid;

                // Set Play Mode and Volume of the Video
                vf.PlayMode = VideoPlayModePreset.Auto;
                vf.Volume = AudioVolumeMode.Loud;

                // Write the PPTX file to disk
                pres.Save(resultPath, SaveFormat.Pptx);
            }
            //ExEnd:EmbeddedVideoFrame
        }
    }
}